app/models/user/availability_switcher.rb

```diff
@@ -0,0 +1,6 @@
+# frozen_string_literal: true
+class User::AvailabilitySwitcher
+  def self.reset
```

Comment:
Two thoughts on this class: 

1. I suggest naming a method like this with a bang (`!`) at the end: `reset!` It's a Ruby convention to name "dangerous" method like that (oftentimes methods have a non-bang and a bang version; a "less-dangerous" non-bang version doesn't really apply here but have a look at the stdlib `String` documentation for a couple of good examples on that).
2. Since the name of the class ends in _Switcher_, it is suggested that there is a reverse action (that is: switching back and forth between two states). The sole action in this class only sets all values to `false`, though. It may sound like nit-picking, but naming is very important (if not amongst the most important things): I suggest naming  this class `User::ResetsAvailability` or something along the lines of the. In order to not re-iterate the name of the class in the one public API method (which would look like `User::ResetsAvailability.reset!`) I suggest the common interface for lamba-/proc style callables: `.call` (this time without a bang – confusing, I know :wink:)